mlflow/tracking/client.py

```diff
@@ -910,6 +914,55 @@ def log_artifacts(self, run_id, local_dir, artifact_path=None):
         """
         self._tracking_client.log_artifacts(run_id, local_dir, artifact_path)
 
+    @contextlib.contextmanager
```

Comment:
BTW, what do you think about pulling these artifact logging utils into a private module, e.g. under `mlflow.artifacts._logging_utils` and actually defining `log_text` there as a static function, and then exposing it via the `MlflowClient` here via a single-line call to `mlflow.artifacts._logging_utils.log_text`? The advantage I think is that we can keep this file a bit more concise, especially as we add logging logic for more artifact types like images etc that may be a bit more complex

Reply:
Happy to have the refactor be in a follow-up PR or in this one, lmk what you think

Reply:
To summarize offline discussion, this actually seems fine - it's hard to define `log_text` itself as a static function due to the dependency on the MlflowClient's `log_artifact`function.

From offline discussion:


> If anything in the future we could consider having utils for logging an in memory object to a pre-specified file path, & that could eventually become the pluggable interface for logging custom object types. You implement "log object type X to a specified file" and then MLflow knows how to use that method to log the object to a tempfile & call log_artifact on it. But I don't think we need to do that just yet